### ISSUE TITLE
[feat] add `allowConstructors` option to `no-unstable-default-props` rule

### ISSUE THREAD

**Issue by christopher-buss** [Rel1cx/eslint-react#1312]
### Problem Description

The `no-unstable-default-props` rule currently flags **all** constructor calls and factory methods in default props as potentially unstable. This is correct for most JavaScript, but some frameworks provide constructors/factories that create **value-based primitives** rather than referential objects.

```tsx
// Currently flagged, but doesn't cause issues in some frameworks
function MyComponent({ position = new Vector3(0, 0, 0) }: { position?: Vector3 }) {
  useEffect(() => {
    console.log("Position:", position);
  }, [position]); // No infinite loop - Vector3 compared by value

  return <div />;
}
```

### Alternative Solutions

**ESLint disable comments** - Too manual, doesn't scale, reduces visibility of real violations

**Disable rule entirely** - Loses protection against actual unstable props

**Module-scoped constants** - Unnecessarily verbose for value types, still need to bypass rule

**Maintain fork** - High maintenance burden, loses upstream updates

None of these solutions adequately address the false positive problem.

### Feature Details

**Feature name:** `safeDefaultProps` option

Add an optional array to specify identifier names (or regex patterns) that are safe to use in default props. When a default prop expression references one of these identifiers, it's excluded from the rule's checks.

**Configuration:**
```typescript
{
  safeDefaultProps?: readonly string[];  // Identifier names or regex patterns
}
```

**Implementation:** Simple identifier-based matching:
- Extract the primary identifier from the default value expression:
  - `new Vector3()` → extract "Vector3"
- Convert config strings to regex using `toRegExp` utility (same as `component-name` rule)
- Check extracted identifier against patterns
- If matched, skip the rule check

**Tradeoffs:** This approach trusts the entire API surface of the allowed identifier (e.g., allowing `vector` permits `vector.create()`, `vector.anyMethod()`, etc.). This is acceptable for framework-provided primitives with controlled APIs.

### Examples

**Basic usage:**
```json
{
  "@eslint-react/no-unstable-default-props": ["error", {
    "safeDefaultProps": ["Vector3", "Color3", "^Immutable.*"]
  }]
}
```

```tsx
// ✅ Allowed
function Component({ pos = new Vector3(0, 0, 0) }) {}

// ❌ Still fails - not in allowlist
function Component({ obj = new CustomClass() }) {}
function Component({ items = [] }) {}
```

**Default behavior (no config):**
```tsx
// ❌ All constructors flagged (current behavior preserved)
function Component({ pos = new Vector3() }) {}
```

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users

**Comment by christopher-buss** [comment]
I am willing to create a PR for this! I just thought I'd link an issue with the motivations for such a feature as well 😄
